src/cli/cli.cpp

```diff
@@ -4234,12 +4234,41 @@ template <> otError Interpreter::Process<Cmd("partitionid")>(Arg aArgs[])
 {
     otError error = OT_ERROR_INVALID_COMMAND;
 
+    /**
+     * @cli partitionid
+     * @code
+     * partitionid
+     * 4294967295
+     * Done
+     * @endcode
+     * @par api_copy
+     * #otThreadGetPartitionId
+     * @par
+     * Get the Thread Network Partition ID.
+     */
     if (aArgs[0].IsEmpty())
     {
         OutputLine("%lu", ToUlong(otThreadGetPartitionId(GetInstancePtr())));
         error = OT_ERROR_NONE;
     }
 #if OPENTHREAD_CONFIG_REFERENCE_DEVICE_ENABLE
+    /**
+     * @cli partitionid preferred (partitionid) (get,set)
+     * @code
+     * partitionid preferred
+     * 4294967295
+     * Done
+     * @endcode
+     * @code
+     * partitionid preferred 0xffffffff
+     * Done
+     * @endcode
+     * @cparam partitionid preferred @ca{partitionid}
+     * @sa otThreadGetPreferredLeaderPartitionId
+     * @sa otThreadSetPreferredLeaderPartitionId
+     * @par
+     * Note: OPENTHREAD_CONFIG_REFERENCE_DEVICE_ENABLE is required.
```

Comment:
This doesn't need to be a note for our docs, but put the flag in code font

`OPENTHREAD_CONFIG_REFERENCE_DEVICE_ENABLE` is required.